tests/core/test_processor.py

```diff
@@ -1881,3 +1881,20 @@ async def test_processor_fetch_full_tracker_with_initial_session_existing_tracke
     tracker = await default_processor.fetch_full_tracker_with_initial_session(sender_id)
     assert tracker.sender_id == sender_id
     assert all([event in expected_events for event in tracker.events])
+
+
+async def test_sender_id_in_logs(
```

Comment:
We should also have this test for the changes made in `memoization.py` and `rule_policy.py` as well.